src/resolvers/resolver.ts

```diff
@@ -1,31 +1,67 @@
 import mongoose from "mongoose";
 import { Profile, User } from "../models/user";
+import bcrypt from "bcryptjs";
+import * as jwt from "jsonwebtoken";
+
+const SECRET = process.env.SECRET || "test_secret";
 
 const resolvers = {
   Query: {
-    hello: () => "Hellooo, welcome to your Graphql server",
-    async getAllUsers() {
-      const users = await User.find({});
-      return users;
-    },
-    async getAllProfiles() {
+    async getAllProfiles(_: any, _args: any, context: { userId: any }) {
+      if (!context.userId) throw new Error("Unauthorized");
```

Comment:
What part of the code catches this error when thrown?

Reply:
am focusing on authentication not profile
may be such issue will be resolved later